nixos/modules/security/wrappers/wrapper.c

```diff
@@ -161,22 +162,25 @@ static int make_caps_ambient(const char *selfPath)
 
 int main(int argc, char * * argv)
 {
-    // I *think* it's safe to assume that a path from a symbolic link
-    // should safely fit within the PATH_MAX system limit. Though I'm
-    // not positive it's safe...
-    char selfPath[PATH_MAX];
-    int selfPathSize = readlink("/proc/self/exe", selfPath, sizeof(selfPath));
+    int selfExe = open("/proc/self/exe", O_PATH | O_CLOEXEC | O_NOFOLLOW);
+    assert(selfExe != -1);
+    struct stat st;
+    assert(fstat(selfExe, &st) != -1);
+    size_t selfPathCap = st.st_size + 1;
+    // Purposefully leak this, it's not going to be around for long...
```

Comment:
By "it's not going to be around for long..." you mean that the operating system will clean this up once the program exits? What is the purpose of purposefully leaking (sorry if I'm being thick)?

Reply:
@ixmatus Our last use of `selfPath` is just before `execve`. If that succeeds, the memory will be freed. If it fails, it will be freed when we exit a few lines later. The only reason to explicitly call `free` would be to satisfy humans checking for good coding practices, and the explicit comment is easier to find than scanning for the matching `free` IMO. But happy to throw in a `free` if that would be better.

Reply:
The "purposefully" bit is just to say "I know this is leaked, it's not an error" for reviewers :)

Reply:
@shlevy oh I understand. I think being explicit with a `free` would be clearer than being implicit and explaining that with a comment.